river_dl/postproc_utils.py

```diff
@@ -143,9 +143,13 @@ def prepped_array_to_df(data_array, dates, ids, col_names, spatial_idx_name='seg
     :return:[pd dataframe] df with cols
     ['date', 'seg_id_nat', 'temp_c', 'discharge_cms]
     """
-    data_array = data_array.flatten()
+    num_out_vars = data_array.shape[2]
+    flat_data = [data_array[:, :, i].flatten() for i in range(num_out_vars)]
```

Comment:
Nice!  The one thing is that GraphWaveNet ends up in shape `[nbatches, nsegs, seq_len, preds]`.  Could we make this model agnostic by changing it to 
```suggestion
    num_out_vars = data_array.shape[-1]
    flat_data = [data_array[..., i].flatten() for i in range(num_out_vars)]
```
Pretty sure this will work with a quick check
<img width="606" alt="image" src="https://user-images.githubusercontent.com/33098707/155608461-3de9b943-9eae-40ad-886b-54f3bb609571.png">

Reply:
Nice!